genic/params.c

```diff
@@ -51,6 +50,7 @@ create_parameters()
     param_declare_int(ps, "NumWriters", OPTIONAL, 0, "");
 
 
+    param_declare_int(ps, "Ngrid", OPTIONAL, 0, "Ignored.");
```

Comment:
I suggest we use Ngrid in GenIC, which controls the particle grid.

Because Nmesh is used in Gadget for the long range resolution, and it is usually different from the particle number. It is supprising to see Nmesh in Gadget and Nmesh in GenIC are different.

Reply:
I usually set Nmesh in both Gadget and GenIC to 2x NParticle! If you set Nmesh = NParticle in GenIC you get odd systematics near the grid scale, where the delta function from the particle grid aliases to lower k.

Reply:
This only occurs with small Nmesh? And moving particles to the center of cell could have fixed it (cic window drops to zero.)

Reply:
No, because after particle displacement the particles are not aligned to cells. This isn't a big effect, but it should happen with all Nmesh. You care only if you are trying to get as accurate as possible power spectra.

Reply:
It is only suppressed by the LCDM power spectrum then (less small scale power -> less aliasing).

Reply:
I see now why the confusion - in GenIC Nmesh controls the number of particles while Ngrid controlled the size of the FFT grid. In S-GenIC I use Nmesh for the size of the FFT grid and N{CDM|Baryon|Neutrino} for the particle number.

How about I file a PR report to add NumParticle as an extra parameter to GenIC, which controls the number of particles? Then Nmesh is the same as in Gadget?

Reply:
The particle is initially on a grid -- that's why I thought Ngrid is a good name -- NumParticle sounds like Ngrid ** 3. Another choice is Nc (number of mass cells per side?), which is used in `cola_halo and fastpm`

Reply:
Good point. Let me bring Ngrid back - then we preserve backwards compatibility as well.